### PR TITLE
Add godoc, golint fixes

### PR DESCRIPTION
**Pull request by sahil-lakhwani**
Adds in code documentation wherever needed, in accordance with godoc.
Modifies code to meet go-lint standards

### CODE REVIEW

File: samples/echobot/doc.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+This methods authenticates the payload, parses the request and returns an Activity value.
+	activity, err := adapter.ParseRequest(ctx, req)
+
+The Activity is then passed to `adapter.ProcessActivity` with the hanlder created to process
+the activity as per the hanlder functions and send the response to the connector service.
+	err = adapter.ProcessActivity(ctx, activity, customHandler)
+
+In case of no error, this web responds with a 200 status
+*/
+package main
```
Comment (PrasadG193) [inline samples/echobot/doc.go]: @sahil-lakhwani We can create a README in this repo and put the documentation there.

Comment (sahil-lakhwani) [reply samples/echobot/doc.go]: Hey, @PrasadG193 Yes we can do that, I chose this so that generated go doc will have the details in it's format.

Comment (PrasadG193) [reply samples/echobot/doc.go]: Let's keep both.

File: connector/auth/claim_identity.go [hunk 2]
```diff
@@ -1,36 +1,45 @@
...
 func NewClaimIdentity(claims map[string]interface{}, isAuth bool) ClaimsIdentity {
 	return &DefaultClaimIdentity{claims, isAuth}
 }
 
+// GetClaimValue returns value for a specified property of a claim
 func (ci DefaultClaimIdentity) GetClaimValue(cType string) string {
 	return ci.claims[cType].(string)
 }
 
+// IsAuthenticated reutrns if the Claim is authenticated
```
Comment (PrasadG193) [inline connector/auth/claim_identity.go]:
```suggestion
// IsAuthenticated returns if the Claim is authenticated
```

File: connector/auth/emulator_validation.go [hunk 3]
```diff
@@ -42,5 +43,6 @@ func IsTokenFromEmulator(authHeader string) bool {
 	return true
 }
 
+// EmulatorTokenValidator provides functionality to check if a request is from an emulator
 type EmulatorTokenValidator struct {
 }
```
Comment (PrasadG193) [inline connector/auth/emulator_validation.go]: We should probably remove this file if we are not using it

File: connector/client/doc.go [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/*Package client provides a HTTP client for calls to the connector service.
+The configuration needed for the call are defined by struct Config
+*/
+package client
```
Comment (PrasadG193) [inline connector/client/doc.go]: Please add newline

File: connector/auth/doc.go [hunk 5]
```diff
@@ -0,0 +1,5 @@
+/*Package auth that provides authentication properties and functionalities for the connector service.
```
Comment (PrasadG193) [inline connector/auth/doc.go]:
```suggestion
/*Package auth provides authentication properties and functionalities for the connector service.
```

File: connector/auth/doc.go [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+TokenValidator provides the functionality for authenticating a JWT token in a received request
```
Comment (PrasadG193) [inline connector/auth/doc.go]: Please add newline in between so that it will appear on separate lines on godoc

File: connector/auth/doc.go [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+Others provide the basic strcutures needed.
```
Comment (PrasadG193) [inline connector/auth/doc.go]:
```suggestion
Others provide the basic structures needed.
```

File: connector/auth/claim_identity.go [hunk 8]
```diff
@@ -1,36 +1,45 @@
...
+// Claim represent a Claim in a JWT token
 type Claim interface{}
+
+// ClaimsIdentity is the interface to process claims in a JWT token
 type ClaimsIdentity interface {
 	GetClaimValue(string) string
 	IsAuthenticated() bool
 }
 
+// DefaultClaim implements Claim
```
Comment (PrasadG193) [inline connector/auth/claim_identity.go]:
```suggestion
// DefaultClaim is a default implementation for Claim
```

File: connector/auth/jwt_token_validation.go [hunk 9]
```diff
@@ -14,19 +14,24 @@ import (
...
 	Activity   schema.Activity
 	AuthHeader string
 }
 
+// NewJwtTokenValidator return a new TokenValidator value
 func NewJwtTokenValidator() TokenValidator {
 	return &JwtTokenValidator{}
 }
 
+// AuthenticateRequest authetiates received request from connector service
```
Comment (PrasadG193) [inline connector/auth/jwt_token_validation.go]:
```suggestion
// AuthenticateRequest authenticates received request from connector service.
```

File: connector/client/client.go [hunk 10]
```diff
@@ -15,14 +15,18 @@ import (
...
 type Client interface {
 	Post(url url.URL, activity schema.Activity) error
 }
 
+// ConnectorClient implements Client to send HTTP requests to the connector service
 type ConnectorClient struct {
 	Config
 }
 
+// NewClient constructs and retirns a new ConnectorClient with provided configuration
```
Comment (PrasadG193) [inline connector/client/client.go]:
```suggestion
// NewClient constructs and returns a new ConnectorClient with provided configuration
```

Comment (PrasadG193) [reply connector/client/client.go]: Please split the statement into two

File: connector/client/client.go [hunk 11]
```diff
@@ -33,6 +37,8 @@ func NewClient(config *Config) (Client, error) {
 }
 
 // Post a activity to given URL
+// Creates a HTTP POST request with the provided activity as the body and a Bearer token in the header
+// Returns any error as received from the call to connector service
```
Comment (PrasadG193) [inline connector/client/client.go]: Add full stop or empty line in between. On godoc it renders as a single line. This happened at multiple places.

File: samples/echobot/doc.go (hunk 1 above)
Comment (PrasadG193) [reply samples/echobot/doc.go]: Also add steps for starting `ngrok`.

File: core/activity/handler.go [hunk 12]
```diff
@@ -5,23 +5,30 @@ import (
...
 type Handler interface {
 	OnMessage(context *TurnContext) (schema.Activity, error)
 }
 
+// HandlerFuncs is an adaptor to let client program specify as many or
+// as few functions to handle events of the connector service while still implementing
+// Handler.
 type HandlerFuncs struct {
 	OnMessageFunc func(turn *TurnContext) (schema.Activity, error)
 }
```
Comment (PrasadG193) [inline core/activity/handler.go]: We should probably add an example snippet showing how to use handlers.
Something like this: https://godoc.org/k8s.io/client-go/tools/cache#NewInformer

File: samples/echobot/doc.go (hunk 1 above)
Comment (sahil-lakhwani) [reply samples/echobot/doc.go]: Added a README as part of this branch https://github.com/infracloudio/msbotbuilder-go/tree/readme

File: connector/client/doc.go (hunk 4 above)
Comment (sahil-lakhwani) [reply connector/client/doc.go]: Removed the file

File: connector/auth/emulator_validation.go (hunk 3 above)
Comment (sahil-lakhwani) [reply connector/auth/emulator_validation.go]: Delete the file

File: core/activity/handler.go (hunk 12 above)
Comment (sahil-lakhwani) [reply core/activity/handler.go]: Added an example in core/bot_framework_adapter_test.go

File: connector/auth/jwt_token_validation.go [hunk 13]
```diff
@@ -35,6 +41,8 @@ func (jv JwtTokenValidator) AuthenticateRequest(ctx context.Context, activity sc
 		return nil, errors.New("Unauthorized Access. Request is not authorized")
 	}
 
+	// TODO : Add emulator verification.
+	//
 	// if IsTokenFromEmulator(authHeader) {
 	// 	return nil
 	// }
```
Comment (PrasadG193) [inline connector/auth/jwt_token_validation.go]: Please remove this

**Pull request merged**